tests/framework/Optimizers/GeneticAlgorithms/discrete/unconstrained/testGAMinwoRepConvHDSM.xml

```diff
@@ -0,0 +1,152 @@
+      <upperBound>6</upperBound>
+      <strategy>withoutReplacement</strategy>
+    </UniformDiscrete>
+  </Distributions>
+
+  <Optimizers>
+    <GeneticAlgorithm name="GAopt">
+      <samplerInit>
+        <limit>30</limit>
+        <initialSeed>42</initialSeed>
+        <writeSteps>every</writeSteps>
+        <type>min</type>
+      </samplerInit>
+
+      <GAparams>
+        <populationSize>10</populationSize>
+        <parentSelection>rouletteWheel</parentSelection>
+        <reproduction>
+          <crossover type="onePointCrossover">
+            <crossoverProb>0.8</crossoverProb>
+          </crossover>
+          <mutation type="swapMutator">
+            <mutationProb>0.9</mutationProb>
+          </mutation>
+        </reproduction>
+        <fitness type="invLinear">
+          <a>2.0</a>
+          <b>1.0</b>
+        </fitness>
+        <survivorSelection>fitnessBased</survivorSelection>
+      </GAparams>
+
+      <convergence>
+        <HDSM>0.99</HDSM>
+      </convergence>
+
+      <variable name="x1">
+        <distribution>uniform_dist_woRepl_1</distribution>
+      </variable>
+
+      <variable name="x2">
+        <distribution>uniform_dist_woRepl_1</distribution>
+      </variable>
+
+      <variable name="x3">
+        <distribution>uniform_dist_woRepl_1</distribution>
+      </variable>
+
+      <objective>ans</objective>
+      <TargetEvaluation class="DataObjects" type="PointSet">optOut</TargetEvaluation>
+    <Sampler class="Samplers" type="MonteCarlo">MC_samp</Sampler>
+    </GeneticAlgorithm>
+  </Optimizers>
+
+  <Samplers>
+    <MonteCarlo name="MC_samp">
+      <samplerInit>
+        <limit>10</limit>
+        <initialSeed>20021986</initialSeed>
+      </samplerInit>
+      <variable name="x1">
+        <distribution>uniform_dist_woRepl_1</distribution>
+      </variable>
+      <variable name="x2">
+        <distribution>uniform_dist_woRepl_1</distribution>
+      </variable>
+      <variable name="x3">
+        <distribution>uniform_dist_woRepl_1</distribution>
+      </variable>
+    </MonteCarlo>
+  </Samplers>
+
+    <Models>
+      <ExternalModel ModuleToLoad="../../../../../AnalyticModels/optimizing/myLocalSum.py" name="myLocalSum" subType="">
+        <variables>x1,x2,x3,ans</variables>
+      </ExternalModel>
+    </Models>
+
+    <DataObjects>
+      <PointSet name="placeholder"/>
+      <PointSet name="optOut">
+        <Input>x1,x2,x3</Input>
+        <Output>ans</Output>
+      </PointSet>
+      <PointSet name="opt_export">
+        <Input>trajID</Input>
+        <Output>x1,x2,x3,ans,age,batchId,fitness,iteration,accepted,HDSM,conv_HDSM</Output>
+      </PointSet>
+    </DataObjects>
+
+    <OutStreams>
+      <Print name="optOut">
+        <type>csv</type>
+        <source>optOut</source>
+      </Print>
+      <Print name="opt_export">
+        <type>csv</type>
+        <source>opt_export</source>
+        <clusterLabel>trajID</clusterLabel>
+      </Print>
+    </OutStreams>
+</Simulation>
```

Comment:
@Jimmy-INL The description of the tests are all the same. Which one gives you inf? Ps. They don't seem to be run by the regression suite (tests file is not modified.)

Reply:
I know, I just wanted you to use them to compare and see if the -inf is causing issues. Since you fixed this please push the gold and add it to tests if the results make sense. Then I will merge this immediately. Thanks for adding better convergence criteria. I guess once this is tested enough, I might remove the old one.

Reply:
@Jimmy-INL since you created the tests, I would ask you to add the gold files and update the descriptions (since you know better then me what you wanted to test hehe :) ) Is it okay?

Thanks a lot.

Reply:
PS. @mandd @wangcj05  since @Jimmy-INL now pushed in this branch, we will need you to review/approve and merge the MR (I don't think GitHub allows him now to merge this since he is a contributor of the MR)